lib/Dialect/FIRRTL/Transforms/CheckCombCycles.cpp

```diff
@@ -301,7 +301,27 @@ class DummySourceNodeIterator
 public:
   explicit DummySourceNodeIterator(Node node, bool end = false)
       : node(node), connect(end ? node.context->connects.end()
-                                : node.context->connects.begin()) {}
+                                : node.context->connects.begin()) {
+    // Remove all the register self initialization connects.
+    while (connect != node.context->connects.end() &&
+           ((*connect).getSrc() == (*connect).getDest() &&
+            isFromRegOp((*connect).getDest())))
+      connect++;
+  }
```

Comment:
I'm not sure it's accurate. Isn't it possible that register connections appear in the middle of `node.context->connects`?